packages/vite/src/node/plugins/asset.ts

```diff
@@ -187,10 +188,13 @@ export function assetPlugin(config: ResolvedConfig): Plugin {
     generateBundle(_, bundle) {
       // do not emit assets for SSR build
       if (config.command === 'build' && config.build.ssr) {
-        for (const file in bundle) {
+        const assetFiles = Object.keys(bundle).filter(
+          (file) => bundle[file].type === 'asset',
+        )
+        for (const file of assetFiles) {
           if (
-            bundle[file].type === 'asset' &&
-            !file.includes('ssr-manifest.json')
+            !file.includes('ssr-manifest.json') &&
```

Comment:
This was also `.includes` in the original code, but would this be better as `.endsWith`?

Reply:
yeah, that does sound like it'd be better

Reply:
Thanks, I have updated this as well.

Reply:
I think we should align with this

https://github.com/vitejs/vite/blob/394148e1e0e790683370e99b5ae550e108797db6/packages/vite/src/node/ssr/ssrManifestPlugin.ts#L79-L82

instead since the manifest file name can be changed. It's a separate bug though so it can be handled separately if it's a lot of work.